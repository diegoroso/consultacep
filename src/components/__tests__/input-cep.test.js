import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import InputCep from '../input-cep'

Enzyme.configure({ adapter: new Adapter() })

const setup = () => {
  const props = {
    onFetchCep: jest.fn,
    onRef: jest.fn
  }

  const enzymeWrapper = shallow(<InputCep {...props} />)

  return {
    props,
    enzymeWrapper
  }
}

describe('InputCep', () => {
  const { enzymeWrapper } = setup()
  expect(enzymeWrapper).toMatchSnapshot();

  it('Renderiza o input sem quebras', () => {
    const div = document.createElement('div')
    expect(enzymeWrapper.unmount(div));
  })
})
