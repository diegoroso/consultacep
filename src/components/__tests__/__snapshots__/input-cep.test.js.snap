// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InputCep
    onFetchCep={[Function]}
    onRef={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "className": "input-cep",
      "data-cy": "InputMask",
      "mask": "99999-999",
      "maskChar": "",
      "onChange": [Function],
      "onClick": [Function],
      "placeholder": "Ex.: 02047-000",
      "type": "tel",
      "value": "",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "className": "input-cep",
        "data-cy": "InputMask",
        "mask": "99999-999",
        "maskChar": "",
        "onChange": [Function],
        "onClick": [Function],
        "placeholder": "Ex.: 02047-000",
        "type": "tel",
        "value": "",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
